app/views/posts/_editor.haml

```diff
@@ -14,13 +14,18 @@
       - elsif current_user.avatar
         = select_tag :icon_dropdown, options_for_select([[current_user.avatar.keyword, current_user.avatar.id]], current_user.avatar.id), prompt: "No Icon"
     .post-info-text
-      .post-character{class: ('hidden' unless item.character), data: {character_id: item.character_id}}= item.character.try(:name)
+      .post-character{class: ('hidden' unless item.character), data: {character_id: item.character_id || '', alias_id: item.character_alias_id || ''}}
```

Comment:
Pretty sure `nil` in the data gets made into a blank string – I at least didn't have trouble at any point that I tested it with the previous setup – but I'm not 100% sure.